tools/VersionController/Models/VersionMetadataHelper.cs

```diff
@@ -0,0 +1,487 @@
+            Dictionary<string, TypeMetadata> oldTypeMetadataDictionary,
+            Dictionary<string, TypeMetadata> newTypeMetadataDictionary,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            var typeMetadataHelper = new TypeMetadataHelper(oldTypeMetadataDictionary, newTypeMetadataDictionary);
+            foreach (var type in oldTypeMetadataDictionary.Keys)
+            {
+                var cmdletMetadata = new CmdletMetadata()
+                {
+                    NounName = "Common",
+                    VerbName = type
+                };
+
+                if (!newTypeMetadataDictionary.ContainsKey(type))
+                {
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: cmdletMetadata,
+                        severity: 0,
+                        problemId: 0,
+                        description: string.Empty,
+                        remediation: string.Empty);
+                    continue;
+                }
+
+                var oldTypeMetadata = oldTypeMetadataDictionary[type];
+                var newTypeMetadata = newTypeMetadataDictionary[type];
+
+                typeMetadataHelper.CompareTypeMetadata(cmdletMetadata, oldTypeMetadata, newTypeMetadata, issueLogger);
+                typeMetadataHelper.CompareMethodMetadata(cmdletMetadata, oldTypeMetadata, newTypeMetadata, issueLogger);
+                typeMetadataHelper.CompareConstructorMetadata(cmdletMetadata, oldTypeMetadata, newTypeMetadata, issueLogger);
+            }
+        }
+
+        /// <summary>
+        /// Determine what kind of version bump should be applied to the common
+        /// code library. We want to ensure that there were no breaking changes
+        /// made to the common code to preserve backwards-compatibility.
+        /// </summary>
+        /// <returns>Version bump that should be applied to the common code library.</returns>
+        public Version GetVersionBumpForCommonCode()
+        {
+            var outputModuleDirectory = _fileHelper.OutputModuleDirectory;
+            var galleryModuleDirectory = _fileHelper.GalleryModuleDirectory;
+
+            Console.WriteLine("Saving AzureRM.Profile from the PowerShell Gallery to check common code changes. This will take a few seconds.");
+            Version versionBump = Version.PATCH;
+            var issueLogger = _logger.CreateLogger<BreakingChangeIssue>("BreakingChangeIssues.csv");
+            using (PowerShell powershell = PowerShell.Create())
+            {
+                powershell.AddScript("Save-Module -Name AzureRM.Profile -Repository PSGallery -Path " + outputModuleDirectory);
+                var cmdletResult = powershell.Invoke();
+            }
+
+            try
+            {
+                var commonAssemblies = new List<string>
+                {
+                    Path.Combine(outputModuleDirectory, "Microsoft.Azure.Commands.Common.Authentication.Abstractions.dll"),
```

Comment:
Can we derive these somehow from the psd1 requiredAssemblies?  i.e. Microsoft.*Commands*